packages/mui-base/src/useSwitch/useSwitch.ts

```diff
@@ -18,18 +16,20 @@ import { UseSwitchParameters, UseSwitchReturnValue } from './useSwitch.types';
  *
  * - [useSwitch API](https://mui.com/base-ui/react-switch/hooks-api/#use-switch)
  */
-export function useSwitch(props: UseSwitchParameters): UseSwitchReturnValue {
+export function useSwitch(params: UseSwitchParameters): UseSwitchReturnValue {
   const {
     checked: checkedProp,
     defaultChecked,
     disabled,
-    onBlur,
+    name,
     onChange,
-    onFocus,
-    onFocusVisible,
     readOnly,
     required,
-  } = props;
+    inputRef: externalInputRef,
```

Comment:
Is this external `inputRef` prop necessary? I notice this prop isn't part of the main root component `SwitchProps`, should it be?

Reply:
I forgot to expose it from the component, thanks!
IMO devs should be able to have access to the native control if they want to.